docs/components/demo/change-default-network.tsx

```diff
@@ -0,0 +1,103 @@
+import { type Chain, mainnet, sepolia } from "@starknet-react/chains";
+import {
+  StarknetConfig,
+  publicProvider,
+  useAccount,
+  useDisconnect,
+  useNetwork,
+} from "@starknet-react/core";
+import { useEffect, useState } from "react";
+import { availableConnectors } from "../starknetkit";
+import { Button } from "../ui/button";
+import { WalletBar } from "../starknet/bar";
+
+export function ChangeDefaultNetwork() {
+  const [defaultChain, setDefaultChain] = useState<Chain>(sepolia);
+  const chains = [mainnet, sepolia];
+  const provider = publicProvider();
+
+  return (
+    <StarknetConfig
+      chains={chains}
+      provider={provider}
+      connectors={availableConnectors}
+      defaultChain={defaultChain} // Propagate the defaultChain option here
+    >
+      <ChangeNetworkInner
+        defaultChain={defaultChain}
+        setDefaultChain={setDefaultChain}
+      />
+    </StarknetConfig>
+  );
```

Comment:
Can you use the `DemoContainer` we already have (which handles wallets etc) and forward the `defaultChainId` to it (and forward it to the `StarknetProvider`).